src/charm.py

```
@@ -569,6 +642,29 @@ def _sync_all_resources(self):
             )
             self.on.refresh_certs.emit()
 
+    def _sync_ext_authz_config(self):
+        policy_manager = self._get_extz_auth_policy_resource_manager()
+        resources = []
+
+        auth_info = self.forward_auth.get_provider_info()
+        if not auth_info or not auth_info.decisions_address:
+            self.ingress_config.publish(ext_authz_service_name="fake_host", ext_authz_port="fake_port")
+            logger.warning("Auth configuration is not ready; missing decisions_address.")
+            policy_manager.reconcile(resources)
+            return
+
+        parsed_url = urlparse(auth_info.decisions_address)
+        service_name = parsed_url.hostname
+        port = parsed_url.port
+
+        self.ingress_config.publish(ext_authz_service_name=service_name, ext_authz_port=str(port))
+        rel = self.model.get_relation(INGRESS_CONFIG_RELATION)
+        if rel and self.ingress_config.is_requirer_ready(rel):
+            provider_name = self.ingress_config.get_ext_authz_provider_name(rel)
```

Say we have an ingress and we add forward_auth.  What is the UX on first reconcile here?  We'll enter this function, send ext_authz_* to istio-k8s, then silently fail this `if` statement.  Do we alert the user that things are only partly configured here?  I don't think we do, but we should

---

This relates to [this comment](https://github.com/canonical/istio-ingress-k8s-operator/pull/47/files#r2017597895) too.  If someone tries to have auth (eg: `if len(auth_relations) > 0` then we:
* should only create a gateway if the auth is ready
* make it really clear to the user when auth is not ready, and why

In general, keep "fail closed" and a fully informed user in mind

---

My gut says breaking `_sync_ext_auth_config` into a few helpers (one to manage auth_info, another to manage the istio-config relation, and maybe another to actually make the authz policy?) will make this easier to test and understand.  Just a suggestion though.  However you implement, keep in mind clarity is really important here since the cost of a mistake is accidentally exposing data